src/snowflake/cli/app/snow_connector.py

```diff
@@ -59,6 +56,40 @@ def connect_to_snowflake(temporary_connection: bool = False, mfa_passcode: Optio
         raise InvalidConnectionConfiguration(err.msg)
 
 
+def _get_connection_details(
+    connection_name: str | None = None,
+    temporary_connection: bool = False,
+    overrides: Dict | None = None,
+):
+    if not temporary_connection:
+        if connection_name is not None:
+            connection_parameters = (
+                get_connection(connection_name)
+                if connection_name
+                else get_default_connection()
+            )
+        else:
+            connection_parameters = get_default_connection()
+    else:
+        connection_parameters = {}
+    if overrides:
+        ctx = click.get_current_context(silent=True)
```

Comment:
I can approve this PR but I don't like that we're mixing logic layers.

IMO when we convert click parameters to `overrides` we should extract ParameterSource there. `overrides` dict does not have to be simple `key -> value` dict, it can be `key -> ParameterOverride(value, source)` dict. And here we should only use already extracted values and sources instead of reaching parameters parsing layer from code responsible for building a connection.